src/test/java/org/apache/sysds/test/applications/EntityResolutionBinaryTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sysds.test.applications;
+
+import org.apache.sysds.test.AutomatedTestBase;
+import org.apache.sysds.test.TestConfiguration;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+
+@RunWith(value = Parameterized.class)
+@net.jcip.annotations.NotThreadSafe
```

Comment:
If you want to avoid the NotThreadSafe, i suggest to make multiple tests that call like the bellow.
I Know it is not a beautiful and concise as the previous, but would make the test non blocking for the other tests.

```java
...

@Test 
public void test1(){
  testScriptEndToEnd(1,1);
}

public void testScriptEndToEnd(int numLshHashTables, int numLshHyperplanes) 
...
```

Reply:
Thats a good idea, maybe that will let us enable the parallel tests again! Will try in the evening.

Reply:
Seems to work. Thanks!